Frameworks/Foundation/NSOperationQueue.mm

```diff
@@ -51,22 +52,16 @@ static inline long _QOSClassForNSQualityOfService(NSQualityOfService quality) {
     }
 }
 
-// Private helper that compares two NSOperations and orders the higher-priority one first
-static NSInteger _compareOperationPriority(NSOperation* op1, NSOperation* op2, void* context) {
-    NSOperationQueuePriority pri1 = op1.queuePriority;
-    NSOperationQueuePriority pri2 = op2.queuePriority;
-    if (pri1 > pri2) {
-        return NSOrderedAscending;
-    } else if (pri1 < pri2) {
-        return NSOrderedDescending;
-    }
-    return NSOrderedSame;
-}
-
 // STL compare for NSOperation*, sorting by queuePriority, putting the greatest priority first
-struct _NSOperation_ComparePrioritySTL {
+// If queue priorities are equal, put the lower queueIndex first
+struct _NSOperation_CompareSTL {
     bool operator()(NSOperation* lhs, NSOperation* rhs) {
-        return lhs.queuePriority < rhs.queuePriority;
+        const NSOperationQueuePriority priLhs = lhs.queuePriority;
+        const NSOperationQueuePriority priRhs = rhs.queuePriority;
+        if (priLhs != priRhs) {
```

Comment:
not equivalent when lhs.queuePriority > rhs.queuePriority and lhs _queueIndex > rhs _queueIndex